tests/test_wallet_wallet_recovery.py

```diff
@@ -40,39 +41,94 @@
 
+@dataclass
+class TestEnv:
+    """Test environment."""
+
+    tmp_path: Path
+    password: str
+    operate: OperateApp
+    wallet_manager: MasterWalletManager
+    keys_manager: KeysManager
+    backup_wallet: str
+    backup_wallet2: str
 
-# TODO decide if use KeysManager method instead.
-def create_crypto(ledger_type: LedgerType, private_key: str) -> Crypto:
-    """create_crypto"""
-    with tempfile.NamedTemporaryFile(mode="w", delete=True) as tmp_file:
-        tmp_file.write(private_key)
-        tmp_file.flush()
-        if ledger_type == LedgerType.ETHEREUM:
-            crypto = EthereumCrypto(private_key_path=tmp_file.name)
-        else:
-            raise NotImplementedError()
-    return crypto
+
+def _create_wallets(wallet_manager: MasterWalletManager) -> None:
+    for ledger_type in LEDGER_TO_CHAINS:
+        wallet_manager.create(ledger_type=ledger_type)
+
+
+def _create_safes(wallet_manager: MasterWalletManager, backup_owner: str) -> None:
+    for ledger_type, chains in LEDGER_TO_CHAINS.items():
+        wallet = wallet_manager.load(ledger_type=ledger_type)
+        for chain in chains:
+            tenderly_add_balance(chain, wallet.address)
+            tenderly_add_balance(chain, backup_owner)
+            wallet.create_safe(
+                chain=chain,
+                backup_owner=backup_owner,
+            )
+
+
+@pytest.fixture
+def test_env(tmp_path: Path, password: str) -> TestEnv:
+    """Sets up a test environment."""
+    operate = OperateApp(
+        home=tmp_path / OPERATE_TEST,
+    )
+    operate.setup()
+    operate.create_user_account(password=password)
+    operate.password = password
+    wallet_manager = operate.wallet_manager
+    wallet_manager.setup()
+    keys_manager = KeysManager(
+        path=operate._path / KEYS_DIR,  # pylint: disable=protected-access
+        logger=LOGGER,
+    )
+    backup_wallet = keys_manager.create()
+    backup_wallet2 = keys_manager.create()
+
+    assert backup_wallet != backup_wallet2
+
+    _create_wallets(wallet_manager=wallet_manager)
+    _create_safes(
+        wallet_manager=wallet_manager,
+        backup_owner=backup_wallet,
+    )
+
+    # Logout
+    operate = OperateApp(
+        home=tmp_path / OPERATE_TEST,
+    )
+
+    return TestEnv(
+        tmp_path=tmp_path,
+        password=password,
+        operate=operate,
+        wallet_manager=wallet_manager,
+        keys_manager=keys_manager,
+        backup_wallet=backup_wallet,
+        backup_wallet2=backup_wallet2,
+    )
 
 
-# TODO enable test once Tenderly RPCs are set up on CI.
-@pytest.mark.skipif(RUNNING_IN_CI, reason="Skip test on CI.")
 class TestWalletRecovery:
     """Tests for wallet.wallet_recoverey_manager.WalletRecoveryManager class."""
 
+    @pytest.fixture(autouse=True)
+    def _patch_rpcs(self, monkeypatch: pytest.MonkeyPatch) -> None:
+        monkeypatch.setattr("operate.ledger.DEFAULT_RPCS", TESTNET_RPCS)
```

Comment:
We need to patch here to use Tenderly sendbalance methods: we need a tesntet here, not mainnet.